src/qnetvo/network_nodes.py

```diff
@@ -0,0 +1,250 @@
+class NetworkNode:
+    """A quantum network node.
+
+    :param num_in: The number of discrete classical inputs that the node accepts.
+    :type num_in: int
+
+    :param num_out: The number of classical outputs for the node.
+    :type num_out: int
+
+    :param wires: The wires on which the node operates.
+    :type wires: list[int]
+
+    :param cc_wires: The classical communication wires input to the node.
+    :type cc_wires: list[int]
+
+    :param cc_wires_out: The classical communication wires to output measurement results on.
+    :type cc_wires_out: list[int]
+
+    :param ansatz_fn: A `PennyLane quantum circuit function <https://docs.pennylane.ai/en/stable/introduction/circuits.html>`_
+        called either as ``circuit(settings, wires)``, or as ``circuit(settings, wires, cc_wires)`` where
+        ``settings`` is an *array[float]* of length ``num_settings`` and ``cc_wires`` contains the measurement results
+        received from upstream nodes.
+    :type ansatz_fn: function
+
+    :param num_settings: The number of settings parameterizing the ``ansatz_fn`` circuit.
+    :type num_settings: int
+
+    :returns: An instance of the ``NetworkNode`` class.
+
+    **Calling a Network Node's Ansatz Function:**
+
+    Given an instantiated network node, ``node = NetworkNode(*args)``, its ansatz quantum circuit functtion
+    can be called as ``node(settings, cc_wires)`` or, if ``node.cc_wires==[]``, the node can be called
+    as  ``node(settings)`` if ``cc_wires=[]``.
+
+    **Attributes:**
+
+    All inputs are stored as class attributes under the same name, *e.g.*, the ``num_in`` argument
+    is stored as ``node.num_in``.
+    """
+
+    def __init__(
+        self,
+        num_in=1,
+        num_out=1,
+        wires=[],
+        cc_wires=[],
+        cc_wires_out=[],
+        ansatz_fn=None,
+        num_settings=0,
+    ):
+        self.num_in = num_in
+        self.num_out = num_out
+
+        self.wires = wires
+        self.cc_wires = cc_wires
```

Comment:
I do not think so, unless there is a mutation side-effect with `__call__` that I'm not aware of...

As I understand it, in the `NetworkNode.__call__` function we input `cc_wires` as the measurement results, which are distinct from the `self.cc_wires`. Perhaps the naming could be altered as `self.cc_wires -> self.cc_wire_ids` to avoid confusion.